File: src/arrow/read-file.py
#!/usr/bin/env python

import os
import pyarrow as pa

times = int(os.getenv('TIMES', 1))

for time in range(times):

    reader = pa.open_file('/tmp/python.arrow')
    schema = reader.schema

    for i in range(reader.num_record_batches):
        record_batch = reader.get_batch(i)
        if times > 1:
            break
        print('=' * 48)
        print(f'record-batch[{i}]:')
        for j, column in enumerate(schema):
            values = record_batch.column(j).to_numpy()
            print(f'  {column.name}: {record_batch.column(j).to_numpy()}')

File: src/arrow/read-file.rb
#!/usr/bin/env ruby

require 'arrow'
require 'pry'

times = ENV.fetch('TIMES', 1).to_i

times.times do
  Arrow::MemoryMappedInputStream.open('/tmp/python.arrow') do |input|
    reader = Arrow::RecordBatchFileReader.new(input)
    fields = reader.schema.fields
    reader.each_with_index do |record_batch, i|
      next if times > 1

      puts '=' * 48
      puts "record-batch[#{i}]:"
      fields.each do |field|
        values = record_batch.map do |record|
          record[field.name]
        end
        puts "  #{field.name}: #{values.inspect}"
      end
    end
  end
end
